src/wallet/wallet.cpp

```diff
@@ -5404,6 +5405,16 @@ bool CMerkleTx::IsLockedByInstantSend() const
     return instantsend.IsLockedInstantSendTransaction(GetHash()) || llmq::quorumInstantSendManager->IsLocked(GetHash());
 }
 
+bool CMerkleTx::IsChainLocked() const
+{
+    AssertLockHeld(cs_main);
+    BlockMap::iterator mi = mapBlockIndex.find(hashBlock);
+    if (mi != mapBlockIndex.end() && mi->second != nullptr) {
```

Comment:
Why would mi->second be `nullptr`?

Reply:
Just a regular safety belt. We still have quite a few places with `mapBlockIndex[somehash]` which can potentially insert an entry if we screw some checks up somewhere (should really get rid of these `[]` some day). Won't hurt to double-check imo.

Reply:
Ah ok, was already worried that I misunderstood `mapBlockIndex` when I used it in various places. An easy way to get rid of the `[]` is to use `.at()` as a replacement as it throws an exception when you try to access a non-existent item, instead of creating an new item as with `[]`. Better to crash early in such cases.